packages/block-library/src/template-part/edit/category-panel.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+import { PanelBody, SelectControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+export default function TemplatePartCategoryPanel( { postId } ) {
+	const [ area, setArea ] = useEntityProp(
+		'postType',
+		'wp_template_part',
+		'area',
+		postId
+	);
+
+	return (
+		<PanelBody title={ __( 'Category' ) }>
+			<SelectControl
+				label={ __( 'Select template part category' ) }
+				labelPosition="top"
```

Comment:
Any idea why this is not working? 🤔 The label is still positioned on the side.

Reply:
I hope you don't mind a pushed a patch here: https://github.com/WordPress/gutenberg/pull/29159/commits/ffa154b2979fac2a3c1b9bb2c7c6e7fcb6d47b76

I'm not sure if this is the right way of handling this though, as it seems this PR: https://github.com/WordPress/gutenberg/pull/28609 introduced this 'conflict' (? 🤔 ). @sarayourfriend @ItsJonQ any feedback here?

Reply:
`Root` correctly applies the CSS based on the `labelPosition`:
![image](https://user-images.githubusercontent.com/2256104/108687741-5d89fb00-74f7-11eb-8906-ae89673c6145.png)

But for some reason a CSS class (which seems to be the `Flex` component) overrides it:
![image](https://user-images.githubusercontent.com/2256104/108687805-72ff2500-74f7-11eb-9296-154dec246054.png)


This is interesting because `Flex`'s styled CSS class should come BEFORE `Root`. `Root` extends `Flex`:
https://github.com/WordPress/gutenberg/blob/0105c9a2312272d768f1ecb77c0b9e6dca77d0a0/packages/components/src/input-control/styles/input-control-styles.js#L45-L52

Oh, and it works fine in Storybook. So it's something with the Editor itself?

Reply:
I think the issue is that we're using two different `styled` functions and the specificity of the Root selector is less than the base Flex selector. There's one that comes from `@wp-g2/styles` that I think using would fix this. @ItsJonQ what's the intended solution in this situation?

Reply:
As @sarayourfriend has pointed out, the issue is indeed caused by the 2 ways the CSS-in-JS styles are being compiled. Something we need to smooth out for the Component System integration.

> Do we actually need the extra specificity? I wonder what would happen if we did without it.

We do. Even before the next Component System (G2), we had to resort to hacks like `&&&` for certain components just so they would render correctly:

https://github.com/WordPress/gutenberg/blob/master/packages/components/src/input-control/styles/input-control-styles.js#L184

My intuition tells me that we need to adjust some props in the `input-control/input-base.js` to ensure it plays nicely with the new `Flex` component (rather than a CSS fix).

Reply:
Update! I have a working solution. It involves remapping props from `input-control` to feed into the new `flex` component. Will create a PR shortly

Reply:
Created a PR here :)
https://github.com/WordPress/gutenberg/pull/29226